### PR TITLE
Issue 479

### PR DESCRIPTION
**Pull request by mnmelo**
There. I think this fixes it, and we'd become independent of having numpy pre-installed (let's see what travis says...)

We'd still face the problem that our dependencies (`biopython`) suffer from the same problem. If this works for us I can bring it up with them, and see if we all get fixed.

### CODE REVIEW

**Comment by jbarnoud** [comment]
I tried on a clean virtualenv. It worked well; biopython was not an issue as they provide a wheel.

File: package/setup.py [hunk 1]
```diff
@@ -131,9 +120,54 @@ def get(self, option_name, default=None):
...
+    def include_dirs(self):
+        if not self._mda_include_dirs:
+            for item in self._mda_include_dir_args:
+                try:
+                    self._mda_include_dirs.append(item()) #The numpy callable
+                except TypeError:
+                    self._mda_include_dirs.append(item) 
+        return self._mda_include_dirs
+
+    @include_dirs.setter
```
Comment (richardjgowers) [inline package/setup.py]: Is this correct? Setting silently does nothing?  Is this so the property is only created when it is first accessed?

**Comment by mnmelo** [comment]
Yes, that's as intended: the initialization of the base `Extension` class now gets no `include_dirs` arg, and so it attempts to set `self.include_dirs` to the empty list. This let's it believe it succeeded.

I'll check whether we can do away with it (not sure if the`@property` decorator is enough to protect the `self.include_dirs` attribute).

**Comment by mnmelo** [comment]
I made the setter decorator a bit less hackish (at least now it does something). It also streamlines the `MDAExtension` class somewhat.
I can't do away with it entirely because then the initialization code of the `Extension` base class borks when trying to assign to `self.include_dirs`.

And now travis is failing in the conda setup stage!? I think travis is broken...

**Comment by mnmelo** [comment]
Ok, it's passing now, and as @jbarnoud (thanks for testing!) noted, we needn't even worry about dependencies.

**Comment by mnmelo** [comment]
So, shall we merge this?

**Pull request merged**